src/shared/utils/analytics.ts

```diff
@@ -1,44 +1,12 @@
 import { Analytics, AnalyticsBrowser } from '@segment/analytics-next';
-import * as Sentry from '@sentry/react';
-import { Integrations } from '@sentry/tracing';
 
-import { IS_PROD_ENV, IS_TEST_ENV, SEGMENT_WRITE_KEY, SENTRY_DSN } from '@shared/environment';
+import { IS_TEST_ENV, SEGMENT_WRITE_KEY } from '@shared/environment';
 import { logger } from '@shared/logger';
 
-import { userHasAllowedDiagnosticsKey } from './storage';
+const analyticsClient = new Promise<null | Analytics>(resolve => {
+  if (IS_TEST_ENV || !SEGMENT_WRITE_KEY) resolve(null);
```

Comment:
Wouldn't it be more accurate to reject in this instance? This means that `Promise.then` is guaranteed to be the client, and doesn't require a null check.

Though, similarly to the docs, consuming this synchronously would be cleaner.